Sources/AppcuesKit/Presentation/Extensions/View+Appcues.swift

```diff
@@ -110,12 +110,15 @@ extension View {
     }
 
     func applyAllAppcues(_ style: AppcuesStyle) -> some View {
-        self
-            .applyForegroundStyle(style)
-            .applyInternalLayout(style)
-            .applyBackgroundStyle(style)
-            .applyBorderStyle(style)
-            .applyExternalLayout(style)
+        // Using `AnyView` here drastically improves memory and CPU usage
+        AnyView(
+            self
+                .applyForegroundStyle(style)
+                .applyInternalLayout(style)
+                .applyBackgroundStyle(style)
+                .applyBorderStyle(style)
+                .applyExternalLayout(style)
+        )
```

Comment:
curious how you found this approach - is this something others have seen as well?

Reply:
So I was trying to trace backwards what was causing the performance issues... My hack-n-slash approach of commenting different bits of code to see what changed led me to the `.applyBackgroundStyle(style)` part in particular.
My interpretation is that all the conditionals we have to apply the styling creates a massively complex type definition that has SwiftUI somehow recomputing _something_. Type erasing with `AnyView` seems to put that to rest. I'd love to understand more and will definitely be digging more into this.

A few interesting articles:
https://nalexn.github.io/anyview-vs-group/
https://chsxf.dev/2022/08/28/5-tup-why-i-quit-using-swiftui.html